lib/src/components/dropdown/dropdown.dart

```diff
@@ -0,0 +1,440 @@
+  bool isInHeader(
+    Offset headerPosition,
+    Size headerSize,
+    Offset clickPosition,
+  ) {
+    return clickPosition.dx >= headerPosition.dx &&
+        clickPosition.dx <= headerPosition.dx + headerSize.width &&
+        clickPosition.dy >= headerPosition.dy &&
+        clickPosition.dy <= headerPosition.dy + headerSize.height;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      width: _size,
+      child: CompositedTransformTarget(
+        link: _link,
+        child: OverlayPortal(
+          controller: _tooltipController,
+          overlayChildBuilder: (BuildContext context) {
+            return CompositedTransformFollower(
+              link: _link,
+              targetAnchor: Alignment
+                  .bottomLeft, // Align overlay dropdown in its correct position
+              child: Align(
+                alignment: AlignmentDirectional.topStart,
+                child: TapRegion(
+                  onTapOutside: (event) {
+                    final headerBox = menuKey.currentContext!
+                        .findRenderObject()! as RenderBox;
+
+                    final headerPosition = headerBox.localToGlobal(Offset.zero);
+
+                    if (!isInHeader(
+                      headerPosition,
+                      headerBox.size,
+                      event.position,
+                    )) _tooltipController.hide();
+                  },
+                  child: ZetaDropDownMenu(
+                    items: widget.items,
+                    selected: widget.selectedItem.value,
+                    width: _size,
+                    boxType: widget.checkBoxType,
+                    onPress: (item) {
+                      if (item != null) {
+                        widget.onChange(item);
+                      }
+                      _tooltipController.hide();
+                    },
+                  ),
+                ),
+              ),
+            );
+          },
+          child: widget.selectedItem.copyWith(
+            round: widget.rounded,
+            focus: _tooltipController.isShowing,
+            press: onTap,
+            itemKey: menuKey,
+          ),
+        ),
+      ),
+    );
+  }
+
+  double get _size => widget.isLarge ? 320 : 120;
+
+  void onTap() {
+    if (!opened) {
+      _tooltipController.toggle();
+    }
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty<bool>('opened', opened))
+      ..add(
+        DiagnosticsProperty<GlobalKey<State<StatefulWidget>>>(
+          'menuKey',
+          menuKey,
+        ),
+      );
+  }
+}
+
+/// Checkbox enum for different checkbox types
+enum CheckBoxType {
+  /// No checkbox
+  none,
+
+  /// Circular checkbox
+  rounded,
+
+  /// Square checkbox
+  square
+}
+
+/// Class for [ZetaDropdownItem]
+class ZetaDropdownItem extends StatefulWidget {
```

Comment:
I think this widget can be private if it's not used outside of the dropdown?

Reply:
Its being used as the input type for the dropdown menu

Reply:
Sure but does it ever get used separately as it's own component? If it doesn't then developers won't need access to it so we can make it private

Reply:
![image](https://github.com/zebrafed/zeta_flutter/assets/99483750/a50966cf-63b9-4957-a033-dbd7e46ad871)
I'm not sure if this is what you mean, but it is being used like this. Would this still need to be private?

Reply:
However is there any reason why this couldn't just be a regular class and then we can use the data from that class to construct the item widget from inside the component? E.g. have a `ZetaDropdownItem` which has a value, icon and label property, then we can use those properties to create the item widgets. I think that might be better than passing through a list of widgets?